core/server.go

```diff
@@ -294,6 +294,11 @@ func (m *RpcServer) GetEnvironments(cxt context.Context, request *pb.GetEnvironm
 		if request.GetShowTaskInfos() {
 			e.Tasks = tasksToShortTaskInfos(tasks, m.state.taskman)
 		}
+
+		if request.GetShowIntegratedServices() {
+			e.IntegratedServicesData = integration.PluginsInstance().GetEnvironmentsData([]uid.ID{env.Id()})[env.Id()]
+		}
+
```

Comment:
Is there a reason to put this code here instead of combining it with this call from above?
```
integratedServicesEnvsData := integration.PluginsInstance().GetEnvironmentsShortData(m.state.environments.Ids())
```
Depending on the flag, we could call either `GetEnvironmentsShortData` or `GetEnvironmentsData` instead of calling always the first, and optionally overwriting it with the 2nd.